examples/frame_prediction/js/entities/entities.js

```diff
@@ -0,0 +1,76 @@
+/************************************************************************************/
+/*                                                                                  */
+/*     Base entity that predicts its position based on the last update              */
+/*                                                                                  */
+/************************************************************************************/
+game.FramePredictionEntity = me.Entity.extend({
+   
+    init: function(x, y, settings) {
+        this._super(me.Entity, "init", [x, y , settings]);
+    },
+    
+    draw: function(renderer) {
+        // Predict the next position to draw at based on velocity
+        var updateInterval = (me.timer.getTime() - me.timer.lastUpdate) / ( (1000/me.sys.updatesPerSecond) );
```

Comment:
The rendered should be agnostic of timing information, because their purpose is drawing one frame at a time. Not animating.

And I made reference to the multiple viewport ticket previously as an example. The renderer API we have now is pretty much what's necessary to build multiple viewports. It would have to get really hacks if each renderer was doing RAF. Not cool.

The TMX layer classes are interesting, since they can optimize drawing based on the visible rectangle. It sounds like that needs to change for multiple TMX support, too ;p

Anyway, passing the delta time for draws would be ideal for predictive positioning.